backend/internal/handlers/update_handlers.go

```diff
@@ -0,0 +1,97 @@
+func UpdateUsername(c *fiber.Ctx) error {
+    var data map[string]string
+
+    if err := c.BodyParser(&data); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Invalid input"})
+    }
+
+    userID := data["user_id"]
+    username := data["username"]
+
+    var user entity.Account
+    if err := database.DB.First(&user, userID).Error; err != nil {
+        return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
+    }
+
+    user.Username = username
+    if err := database.DB.Save(&user).Error; err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't update username"})
+    }
+
+    return c.JSON(fiber.Map{"message": "Username updated successfully"})
+}
+
+// UpdateEmail updates a user's email
+func UpdateEmail(c *fiber.Ctx) error {
+    var data map[string]string
+
+    if err := c.BodyParser(&data); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Invalid input"})
+    }
+
+    userID := data["user_id"]
+    email := data["email"]
+    confirmEmail := data["confirm_email"]
+
+    if email != confirmEmail {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Emails do not match"})
+    }
+
+    var user entity.Account
+    if err := database.DB.First(&user, userID).Error; err != nil {
+        return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
+    }
+
+    user.Email = email
+    if err := database.DB.Save(&user).Error; err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't update email"})
+    }
+
+    return c.JSON(fiber.Map{"message": "Email updated successfully"})
+}
+
+// UpdatePassword updates a user's password and logs them out
+func UpdatePassword(c *fiber.Ctx) error {
+    var data map[string]string
+
+    if err := c.BodyParser(&data); err != nil {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Invalid input"})
+    }
+
+    userID := data["user_id"]
+    password := data["password"]
+    confirmPassword := data["confirm_password"]
+
+    if password != confirmPassword {
+        return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{"message": "Passwords do not match"})
+    }
+
+    var user entity.Account
+    if err := database.DB.First(&user, userID).Error; err != nil {
+        return c.Status(fiber.StatusNotFound).JSON(fiber.Map{"message": "User not found"})
+    }
+
+    hashedPassword, err := utils.HashPassword(password)
+    if err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't hash password"})
+    }
+
+    user.Password = hashedPassword
+    if err := database.DB.Save(&user).Error; err != nil {
+        return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{"message": "Couldn't update password"})
+    }
+
+    // Log out logic can be added here
+
```

Comment:
Most of the logic for logging out a user is to revoke their token. We will need to add a new function in the token.go file that will revoke the current token (that should also be used for the logout feature). This could also be done in the front end, but we definitely want to expire a user token when login out on the backend too.